### ISSUE TITLE
[BUG] - 404 while access to plugin

### ISSUE THREAD

**Issue by Borch97** [bcombs18/osTicket-Inventory-Manager#73]
**Describe the bug**
I have successfully installed version 1.1.5 by following the installation steps.
Once the plugin is activated, inside the agent panel, in Applications, the Inventory Manager drop-down appears and when I click on it, it takes me to the URL "<IP>/scp/dispatcher.php/inventory/asset/handle" and it gives me a 404 response, telling me it can't find the file.

I have tried to install other versions of the plugin, but the same thing happens to me.

**To Reproduce**
1. Go to 'Applications in Agent panel'
2. Click on 'Inventory Manager'
4. See error 404 at url <IP>/scp/dispatcher.php/inventory/asset/handle

**Expected behavior**
View plugin Dashboard

**Additional context**
OsTicket 1.16.1
PHP: 8.0

**Comment by bcombs18** [comment]
@Borch97 Hello, it sounds like you might be missing the dispatcher.php file. In the root directory of the Inventory Manager plugin, you will find the dispatcher.php file. Move or copy this file to osticket/scp/. Then try clicking the link to the plugin again.

**Comment by Borch97** [comment]
I have already done that step following the installation instructions.

Similarly, all files have the permissions assigned to www-data.

Any other idea?

**Comment by bcombs18** [comment]
Is the /scp/dispatcher.php/inventory/asset/handle link missing osticket/ (or something else if you've set up a custom URI) at the beginning of the URL?

**Comment by Borch97** [comment]
This is my scp folder at this time.

![Screenshot from 2022-09-29 19-42-16](https://user-images.githubusercontent.com/16107693/193104193-ce60c268-958f-4d2b-b678-e2822fca7a9e.png)

My root url is 127.0.0.1:15200, so mu scp path is 127.0.0.1:15200/scp.

**Comment by bcombs18** [comment]
Okay, what happens if you manually go to 127.0.0.1:15200/scp/dispatcher.php/inventory/asset/handle ?

**Comment by Borch97** [comment]
If i manually go to that url, i still get the 404 error.

Is there any extra information I can give you to help you find the problem?

**Comment by Borch97** [comment]
![Screenshot from 2022-09-29 19-52-57](https://user-images.githubusercontent.com/16107693/193107219-16527b76-cc14-4b16-aa57-8d1ce378ae7e.png)

In the php_error.log i have nothing about the plugin.

**Comment by bcombs18** [comment]
Okay, thanks for those. It seems like the plugin isn't building the URL correctly on your osTicket instance for some reason. Can you hover over the Applications tab and right click on Inventory Manager, then select Inspect. Then grab a screenshot of the code it highlights. Here is an example of what I'm looking for.

![image](https://user-images.githubusercontent.com/25602455/193109086-87f67b17-3d1a-474e-bf7e-d0a0b119701c.png)

**Comment by Borch97** [comment]
There you go.

![Screenshot from 2022-09-29 20-07-14](https://user-images.githubusercontent.com/16107693/193109676-36d3df90-0bbd-4f34-b6af-535bdacf99e5.png)

**Comment by Borch97** [comment]
I install it with the both ways, ending with the same result in all cases.

This is my dispatcher.php file at the scp folder.


![Screenshot from 2022-09-29 20-25-35](https://user-images.githubusercontent.com/16107693/193113099-a30d4d07-83be-467e-86b2-ee6ea5ca3dd8.png)

**Comment by bcombs18** [comment]
Okay, It seems something is wrong with that dispatcher.php file. The only way I am able to recreate the error is if I rename/remove the dispatcher.php file from my scp directory. I'm wondering if Apache isn't seeing the file for some reason? Can you try restarting your Apache server and see if that changes anything?

**Comment by Borch97** [comment]
I restarted multiple times without result change. I alsa have a dispatcher.php file in scp/apps. Is that normal? That file is from fresh install.

![Screenshot from 2022-09-29 20-31-42](https://user-images.githubusercontent.com/16107693/193114159-9276a90a-c823-448d-a0ad-e211d088793c.png)

**Comment by bcombs18** [comment]
Yes, that file is normal. The current version of osTicket is missing a file called apps.php which is supposed to route plugins through the apps/dispatcher.php file. The workaround is to put it into the scp directory. The version of that file that comes with the plugin is also slightly modified.

If you go to http://127.0.0.1:15200/scp/dispatcher.php, do you get a 404 error or a message that says "URL not supported"?

**Comment by Borch97** [comment]
I get URL not supported.

**Comment by bcombs18** [comment]
Okay, that sort of narrows down the issue. That file is able to be accessed, but it isn't finding the files in the plugin. What file permissions are set on the plugin directory and sub-directories?

**Comment by Borch97** [comment]
Here plugin root folder and assets folder permissions. All files have the same owner and permissions.

![Screenshot from 2022-09-29 20-47-11](https://user-images.githubusercontent.com/16107693/193117307-15ac046b-1db8-47cd-ac94-98a26fce16b8.png)
![Screenshot from 2022-09-29 20-48-31](https://user-images.githubusercontent.com/16107693/193117328-83c12829-9262-4e26-b4ce-606de28c8322.png)

**Comment by bcombs18** [comment]
Can you send your apache_error.log file? If we see an error like "script 'C:/wamp64/www/osticket/scp/dispatcher.php' not found or unable to stat", then the issue will likely be something in the Apache configuration. It may be looking for a script redirect rather than accessing the files directly.

Was there a particular guide you followed for getting osTicket setup? I'm curious if there might have been some configuration that is preventing access to the plugin.

**Comment by Borch97** [comment]
My configuration includes nginx with php-fpm, not apache. Also, is ubuntu based instalation with docker.

This is my nginx error log file.

![Screenshot from 2022-09-29 21-20-02](https://user-images.githubusercontent.com/16107693/193123155-5a733cc4-857f-45a4-91f0-3ead5877a165.png)

**Comment by Borch97** [comment]
This is my nginx configuration file.

![Screenshot from 2022-09-29 21-22-32](https://user-images.githubusercontent.com/16107693/193123626-7c1b0156-6b92-4f3e-97a8-3c290a6128bb.png)

**Comment by bcombs18** [comment]
What happens if you add the following to your configuration file? I'm not super familiar with nginx so this might not be completely correct.

```
location ~ "^/scp/dispatcher.php/.*$ {
      try_files $uri $uri/ /scp/dispatcher.php;
}

if ($request_uri ~ "^/dispatcher.php(/[^\?]+) {
      set $path_info $1;
}
```

**Comment by bcombs18** [comment]
Also, just as a test, temporarily remove the "deny all" from the "location ~ /include" section and restart your server. Then see if the plugin will load. Since the plugin resides in the include directory, I'm curious if that directive is what is denying access. If the plugin loads with out that part, then we need to add another directive to allow access only to the plugin files.
